File: internal/domain/entity/member.go
package entity

import "time"

type MemberRole string

const (
	RoleMember  MemberRole = "member"
	RoleTrainer MemberRole = "trainer"
	RoleAdmin   MemberRole = "admin"
)

type Member struct {
	ID               int64      `json:"id"                db:"id"`
	FIO              string     `json:"fio"               db:"fio"`
	TelegramUsername string     `json:"telegram_username" db:"telegram_username"`
	TelegramID       int64      `json:"telegram_id"       db:"telegram_id"`
	BirthDate        *time.Time `json:"birth_date"        db:"birth_date"`
	LeftAt           *time.Time `json:"left_at"           db:"left_at"`
	CreatedAt        time.Time  `json:"created_at"        db:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at"        db:"updated_at"`
}

type ClubMember struct {
	ID       int64      `json:"id"        db:"id"`
	ClubID   int64      `json:"club_id"   db:"club_id"`
	MemberID int64      `json:"member_id" db:"member_id"`
	Role     MemberRole `json:"role"      db:"role"`
	JoinedAt time.Time  `json:"joined_at" db:"joined_at"`
}

File: web/admin/src/api/member.ts
import client from './client'

export interface Member {
  id: number
  fio: string
  telegram_username: string
  telegram_id: number
  birth_date?: string
  role: string
  created_at?: string
  updated_at?: string
}

export interface MemberCreate {
  fio: string
  telegram_username: string
  role: string
}

export interface MemberUpdate {
  fio?: string
  telegram_username?: string
  birth_date?: string
}

export default {
  list(clubId: number) {
    return client.get<Member[]>(`/clubs/${clubId}/members`)
  },

  create(clubId: number, data: MemberCreate) {
    return client.post<Member>(`/clubs/${clubId}/members`, data)
  },

  get(id: number) {
    return client.get<Member>(`/members/${id}`)
  },

  update(id: number, data: MemberUpdate) {
    return client.put<Member>(`/members/${id}`, data)
  },

  remove(id: number) {
    return client.delete(`/members/${id}`)
  },

  updateRole(clubId: number, memberId: number, role: string) {
    return client.put(`/clubs/${clubId}/members/${memberId}/role`, { role })
  },
}
